### ISSUE TITLE
Add ignoreSelectors: [] to max-nesting-depth

### ISSUE THREAD

**Issue by emattias** [stylelint/stylelint#4805]
I need an option in `max-nesting-depth` to allow nesting for certain attributes (either specified specifically or with regex). 

Here's an example of what I am talking about:

```scss
.foo {
  &[disabled] { // <- I want to allow this
    opacity: 0.4;
  }
  &.bar { // <- but disallow this.
    color: red;
  }
}
```

**Comment by jeddy3** [comment]
@emattias Thanks for the request and for using the template.

We can add an option, similar to the existing [`ignoreAtRules: []`](https://stylelint.io/user-guide/rules/max-nesting-depth#ignoreatrules-regex-regex-string) option, that targets selectors. You'll then need to craft your own regex to identify the attribute selectors you want to ignore.

I've labelled the issue as ready to implement. Please consider [contributing](https://stylelint.io/contributing) if you have time.

There are [steps on how to add a new option](https://stylelint.io/developer-guide/rules#add-an-option-to-a-rule) in the Developer guide.

**Comment by malsf21** [comment]
Looks like this was in limbo, so tried my hand at an implementation in #4893.
